wagtail/admin/templates/wagtailadmin/generic/index_results.html

```diff
@@ -23,15 +23,25 @@ <h2 role="alert">
 {% endblock %}
 
 {% block no_results_message %}
-    {% if is_searching or is_filtering %}
-        {% blocktrans trimmed with model_name=model_opts.verbose_name_plural asvar no_results_text %}
-            No {{ model_name }} match your query
-        {% endblocktrans %}
-    {% else %}
-        {% blocktrans trimmed with model_name=model_opts.verbose_name_plural asvar no_results_text %}
-            There are no {{ model_name }} to display
-        {% endblocktrans %}
-    {% endif %}
+    {% fragment as no_results_message %}
+        {% if is_searching or is_filtering %}
+            {% blocktrans trimmed with model_name=model_opts.verbose_name_plural asvar no_results_text %}
+                No {{ model_name }} match your query.
+            {% endblocktrans %}
+            {{ no_results_text|capfirst }}
+        {% else %}
+            {% blocktrans trimmed with model_name=model_opts.verbose_name_plural asvar no_results_text %}
+                There are no {{ model_name }} to display.
+            {% endblocktrans %}
+            {{ no_results_text|capfirst }}
+
+            {% if add_url %}
+                {% blocktrans trimmed %}
+                    Why not <a href="{{ add_url }}">add one</a>?
+                {% endblocktrans %}
+            {% endif %}
```

Comment:
Not sure if this is enough context for the translators though. I guess I can also make this an if/else with two translatable blocks, e.g. having
- `"There are no {{ model_name }} to display.` as one block
- `"There are no {{ model_name }} to display. Why not <a href="{{ add_url }}">add one</a>?"` as the other.

Or maybe rename the variable to `add_model_url`? Not sure if that helps much

Reply:
I think I'd go with the "two translatable blocks" solution. I notice that there are various other result templates such as [wagtailredirects/results.html](https://github.com/wagtail/wagtail/blob/main/wagtail/contrib/redirects/templates/wagtailredirects/results.html) that have the same text without the `model_name` variable (and probably need the same permission check, but at least we can consider that out of scope for #11393...) - in those cases it's definitely better to have both sentences in the same string, to allow the "add one" text to be adjusted to fit the model name (e.g. matching gender). It makes sense for the snippet template to follow the same pattern, even if that particular advantage isn't present here.

Reply:
OK, added in 9da5434. Curious what we'd prefer when we eventually refactor those views though (i.e. whether to keep the original block overrides or reuse the generic one to unify the code), but we'll cross that bridge when we come to it.